### DISCUSSION TITLE
How do I upgrade csc to 4.5?

### DISCUSSION THREAD

**Discussion by kevgrig** [jellyfin/jellyfin#10373] [Q&A] [answered]
I'm trying to switch to the `master` branch and I'm on Fedora 37. When I clean and build, I get the following error:

`CSC : error CS9057: The analyzer assembly '/home/kevin/.nuget/packages/idisposableanalyzers/4.0.7/analyzers/dotnet/cs/IDisposableAnalyzers.dll' references version '4.5.0.0' of the compiler, which is newer than the currently running version '4.4.0.0'. [/work/jellyfin/src/jellyfin/src/Jellyfin.Extensions/Jellyfin.Extensions.csproj]`

It looks like the `master` branch depends on idisposableanalyzers 4.7.0:

```
$ grep -r IDisposableAnalyzers *
Directory.Packages.props:    <PackageVersion Include="IDisposableAnalyzers" Version="4.0.7" />
[...]
```

The error above suggests this package depends on csc 4.5 whereas mine is 4.4 (installed through the Fedora package manager):

```
$ dotnet /usr/lib64/dotnet/sdk/7.0.111/Roslyn/bincore/csc.dll --version | head -1
Microsoft (R) Visual C# Compiler version 4.4.0-6.23101.13 (528676cd)
```

**Comment by kevgrig** [comment, marked as answer]
The solution was to manually install the latest .NET 7 SDK: https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#manual-install

And then:

```
$ export DOTNET_ROOT=/work/dotnet/7.0.401/.dotnet
$ export PATH=$DOTNET_ROOT:$DOTNET_ROOT/tool:$PATH
$ dotnet build
[...]
Build succeeded.
    0 Warning(s)
    0 Error(s)
```

**Comment by Shadowghost** [comment]
Just as a heads up this is also the case for Ubuntu 22.04 with packages installed from the ubuntu repositories.

**Comment by MisterMcDuck** [comment]
Just in case someone doesn't want to manually install this on Ubuntu when the apt version should update eventually, downgrading IDisposableAnalyzers to 4.0.4 in Directory.Packages.props will also solve the problem.
